spec/lib/api/2.0/skus-spec.js

```diff
@@ -0,0 +1,360 @@
+                var skus = helper.injector.get('Http.Api.Skus');
+                var fs = helper.injector.get('fs');
+                var req = fs.createReadStream('spec/lib/services/sku-static/pack.tar.gz');
+                var Promise = helper.injector.get('Promise');
+                var promise = new Promise(function(resolve, reject) {
+                    _resolve = resolve;
+                    _reject = reject;
+                });
+                var res = {
+                    status: function(code) {
+                        if(code === 201) {
+                            _resolve(201);
+                        }
+                        _reject(code);
+                        return res;
+                    },
+                    send: sinon.stub(),
+                    json: function(obj) {
+                        return JSON.stringify(obj);
+                    }
+                };
+
+                req.on('open', function() {
+                    skus.skuPackHandler(req, res);
+                });
+
+                return promise.then(function(code) {
+                    expect(code).to.equal(201);
+                    expect(skuService.installPack.called).to.be.true;
+                    expect(skuService.registerPack.called).to.be.true;
+                });
+            });
+
+            it('should process a .tar.gz file with a sku', function() {
+                var skus = helper.injector.get('Http.Api.Skus');
+                var fs = helper.injector.get('fs');
+                var req = fs.createReadStream('spec/lib/services/sku-static/pack.tar.gz');
+                var Promise = helper.injector.get('Promise');
+                var promise = new Promise(function(resolve, reject) {
+                    _resolve = resolve;
+                    _reject = reject;
+                });
+                var res = {
+                    status: function(code) {
+                        if(code === 201) {
+                            _resolve(201);
+                        }
+                        _reject(code);
+                        return res;
+                    },
+                    send: sinon.stub(),
+                    json: function(obj) {
+                        return JSON.stringify(obj);
+                    }
+                };
+
+                skus.skuPackHandler(req, res, 'skuid');
+
+                return promise.then(function(code) {
+                    expect(code).to.equal(201);
+                    expect(skuService.installPack.called).to.be.true;
+                    expect(skuService.registerPack.called).to.be.true;
+                });
+            });
+        });
+
+        describe('DELETE /skus/:id/pack', function () {
+            beforeEach('create node', function () {
+                var waterline = helper.injector.get('Services.Waterline');
+                return waterline.nodes.create({ name: 'sku test node' }).then(function (node_) {
```

Comment:
! You should avoid manipulating the real database in unit-testing, you can turn to use the sinon.stub